### PR TITLE
feat: limit number of events sent during a sync

### LINKED ISSUE #11607: Limit number of events in a sync
Set an upper limit for the number of events to send during a sync. Will protect nodes that are behind from getting massive syncs.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #11607

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowgraphSynchronizer.java [hunk 1]
```diff
@@ -382,6 +388,10 @@ private List<EventImpl> createSendList(
             sendList = eventsTheyMayNeed;
         }
 
+        if (maximumEventsPerSync > 0 && sendList.size() > maximumEventsPerSync) {
+            sendList = sendList.subList(0, maximumEventsPerSync);
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowgraphSynchronizer.java:392]: I think its a lot better if we don't even add them to the list in the first place

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/config/SyncConfig.java [hunk 2]
```diff
@@ -52,4 +53,5 @@ public record SyncConfig(
         @ConfigProperty(defaultValue = "true") boolean filterLikelyDuplicates,
         @ConfigProperty(defaultValue = "3s") Duration nonAncestorFilterThreshold,
         @ConfigProperty(defaultValue = "500ms") Duration syncKeepalivePeriod,
-        @ConfigProperty(defaultValue = "1m") Duration maxSyncTime) {}
+        @ConfigProperty(defaultValue = "1m") Duration maxSyncTime,
+        @ConfigProperty(defaultValue = "100") int maxSyncEventCount) {}
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/config/SyncConfig.java:57]: I think we should start with a really high number, and then lower it when we are able to adequately test this

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowgraphSynchronizer.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowgraphSynchronizer.java:392]: Challenge with that approach is that we gather events out of order, and sort them at the end. It's not obvious to me how to just gather the right amount.

**Review by alittley — DISMISSED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/config/SyncConfig.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/config/SyncConfig.java]: I've disabled this feature by default (0 == no size limit). We can enable it when it gets to the perf environments where we can test it.

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowgraphSynchronizer.java (hunk 1 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowgraphSynchronizer.java:373]: yeah, lets not complicate it now

**Review by lpetrovic05 — APPROVED** [review]

**Pull request merged**